tests/std/tests/P1135R6_barrier/test.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <atomic>
+#include <barrier>
+#include <cassert>
+#include <thread>
+#include <utility>
+
+void test() {
+    std::barrier b(2);
+
+    std::atomic<int> c{0};
+
+    std::thread t1([&] {
+        for (int i = 0; i < 5; ++i) {
+            auto token = b.arrive();
+            b.wait(std::move(token));
+            c.fetch_add(1, std::memory_order_relaxed);
+        }
+    });
+
+    std::thread t2([&] {
+        for (int i = 0; i < 3; ++i) {
+            b.arrive_and_wait();
+            c.fetch_add(1, std::memory_order_relaxed);
+        }
+        b.arrive_and_drop();

```

Comment:
This should test that the "drop" part did something, see example like that I posted in the barrier product code:

```
barrier<some completion that increments a counter> b{2};
jthread t1{[&] {
    b.arrive_and_wait();
}};
b.arrive_and_drop();
b.arrive_and_drop();
assert(completion called twice);
```

Reply:
This example is UB. 
For white-box testing `arrive_and_drop` can only be tested indirectly via called_times.
I have that coverage.

Reply:
Ah, I see, you're right if both arrive_and_drops happen first, then the arrive_and_wait tries to arrive when current is 0.

```
barrier<some completion that increments a counter> b{2};
thread t{[&] {
    b.arrive_and_wait();
}};
b.arrive_and_drop();
t.join();
b.arrive_and_drop();
assert(completion called twice);
```

fixes that but I think the single threaded one I sent you over Discord would also be OK. (And you might have already put that in here as I write this, I haven't checked yet)

Reply:
Looking at this again I think the secnario is now tested.